modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -680,9 +680,11 @@ else if (value instanceof Object[]) {
 
 			if (indexType.equals("keyword")) {
 				document.addKeywordSortable(name, valuesString);
+				document.addKeyword(_getSortableFieldName(name), valuesString);
 			}
 			else {
 				document.addTextSortable(name, valuesString);
+				document.addText(_getSortableFieldName(name), valuesString);
```

Comment:
@p-albuquerque do we still need the `document.addTextSortable(name, valuesString);`?

Reply:
I will check this

Reply:
Without this part the fix doesn't works anymore.

Also, I noticed a difference between the [search field](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/portal-kernel/src/com/liferay/portal/kernel/search/Field.java) created with and without it: 

Without `document.addTextSortable(name, valuesString)` :
![withoutSortable](https://github.com/liferay-forms/liferay-portal/assets/79872134/ca0cf15a-8bdf-4f8d-a709-9d3fc3307797)

With `document.addTextSortable(name, valuesString)` :
![withSortable](https://github.com/liferay-forms/liferay-portal/assets/79872134/89bf3510-918f-4167-927a-a55689edef75)